src/main/java/pl/edu/agh/firecell/storage/Storage.java

```diff
@@ -1,10 +1,4 @@
 package pl.edu.agh.firecell.storage;
 
-import pl.edu.agh.firecell.model.State;
-
-import java.util.Optional;
-
-public interface Storage {
-    void putState(State state, int index);
-    Optional<State> getState(int index);
+public interface Storage extends StateConsumer, StateProvider {
```

Comment:
I think merging back this to interfaces only to have shorter `implements ...`, after we splitted them into separate ones does not makes sense, I mean the existing of `Storage` is not justified.

Reply:
removed `Storage`